### PR TITLE
Spring cleaning

### PR DESCRIPTION
**Pull request by radex**
Various cleanup with many trivial, but breaking changes:

- Refactored Model._preparedState in preparation to new delete API
- removing old unused stuff, refactoring things so that they're loaded lazily, better minifiable, etc.

### CODE REVIEW

File: src/adapters/lokijs/index.js [hunk 1]
```diff
@@ -78,6 +41,21 @@ export type LokiAdapterOptions = $Exact<{
     // This happens if there's another open tab of the same app that's making changes.
     // You might use it as an opportunity to alert user to the potential loss of data
     onDidOverwrite?: () => void,
+    // Called when internal IndexedDB version changed (most likely the database was deleted in another browser tab)
+    // Pass a callback to force log out in this copy of the app as well
+    // (Due to a race condition, it's usually best to just reload the web app)
+    // Note that this only works when not using web workers
+    onversionchange?: () => void,
```
Comment (michalpopek) [inline src/adapters/lokijs/index.js:48]: Any reasons to keep the name lowercase?

File: src/adapters/lokijs/worker/lokiExtensions.js [hunk 2]
```diff
@@ -60,13 +58,7 @@ async function getLokiAdapter(options: LokiAdapterOptions): mixed {
         ? require('lokijs/src/incremental-indexeddb-adapter')
         : require('lokijs/src/incremental-indexeddb-adapter')
       // $FlowFixMe
-      return new IncrementalIDBAdapter({
-        onversionchange: onIndexedDBVersionChange,
```
Comment (michalpopek) [inline src/adapters/lokijs/worker/lokiExtensions.js:64]: Oh I see, this has always been lowercase

**Review by michalpopek — APPROVED** [review]

File: src/adapters/lokijs/worker/lokiExtensions.js (hunk 2 above)
Comment (radex) [reply src/adapters/lokijs/worker/lokiExtensions.js:64]: yes. ugly, but not worth changing IMO. `onversionchange` follows the DOM convention

File: src/Collection/index.js [hunk 3]
```diff
@@ -223,9 +220,4 @@ export default class Collection<Record: Model> {
       idx !== -1 && this._subscribers.splice(idx, 1)
     }
   }
-
-  // See: Database.unsafeClearCaches
-  unsafeClearCache(): void {
```
[the remark this answers is not in the document]
Comment (radex) [reply src/Collection/index.js:228]: not really, this never was meant to be public and I can't imagine anyone actually using it. I'll add it to changelog in my next PR just to be sure

**Pull request merged**